packages/graphql/lib/decorators/resolve-object-type.decorator.ts

```diff
@@ -0,0 +1,15 @@
+import { SetMetadata } from "@nestjs/common";
+
+/**
+ * Designates an instance method as a type resolver for an object type
+ */
+export const RESOLVE_OBJECT_TYPE_METADATA = 'graphql:resolve_object_type';
+export function ResolveObjectType(): MethodDecorator {
+  return (
+    target: Function | Object,
+    key?: string | symbol,
+    descriptor?: any,
+  ) => {
+    SetMetadata(RESOLVE_OBJECT_TYPE_METADATA, true)(target, key, descriptor);
+  };
+}
```

Comment:
I didn't know how to correctly set up `const`s like this so that they end up in the consts file.

This decorator exists to ensure that we wouldn't accidentally opt people into this behavior.  You must annotate your method that'd handle object type resolution explicitly, otherwise Nest will just carry on as it always has with field resolvers.